boards/omote: add support for OMote board

Oppila-ms opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description

<!--
Put here the description of your contribution:
- omote is a lightweight and powerful IOT development board which comes with 32 bit ARM cortex-M3 micro controller and IEEE 802.15.4 radio
-->Features

    • ARM cortex-M3 
    • Transceiver with PCB Antenna
    • on board sensors BMP180,ADXL345 and LDR
    • Programming over BSL
    • User LEDs and Power LEDs
    • User, BSL and reset Buttons 
    • on-board 3.3V and 5V Power output
    • Peripherals- UART, GPIO, I2C and SPI, ADC
    • RF 2.4GHz built-in in CC2538

sales@oppila.in – Oppila Microsystems pvt ltd Banglore

### Testing procedure

<!--
Details steps to test your contribution:
- The tool chain used to build is arm-gcc
        arm-none-eabi-gcc should be installed

The omote should be flashed over USB using the BSL. Bootloader should be unlocked by manually pressing the BSL button and reset button. board can be flashed using following command 

$make BOARD=omote flash 
-->

benpicco commented on boards/OMote/README.md:
```diff
@@ -0,0 +1,104 @@
+### For the CC2538EM (USB CDC-ACM)
+The OMote has built-in support for USB 2.0 USB, Vendor and Product IDs are the following:
+
+  * VID 0x0451
+  * PID 0x16C8
```
If you want to use those you should add

```C
#define INTERNAL_PERIPHERAL_VID         (0x0451)
#define INTERNAL_PERIPHERAL_PID         (0x16C8)
```

to `board.h`

benpicco commented on boards/OMote/Makefile.include:
```diff
@@ -0,0 +1,92 @@
+
+# include common cc2538 includes
+INCLUDES += -I$(RIOTBOARD)/common/cc2538/include
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```
```suggestion
```

drop the newlines

benpicco commented on boards/OMote/README.md:
```diff
@@ -0,0 +1,104 @@
+following goodies:
+
+* ISM 2.4-GHz IEEE 802.15.4 & Zigbee compliant.
+* ISM 868-, 915-, 920-, 950-MHz ISM/SRD Band.
```
I thought the CC2538 is 2.4 GHz only - is there another radio on the board?

benpicco commented on boards/OMote/README.md:
```diff
@@ -0,0 +1,104 @@
+
+More Reading
+============
+1. [Oppila Microsystems OMote website]
```
This is missing the URL

benpicco commented on boards/OMote/Makefile.include:
```diff
@@ -0,0 +1,92 @@
+PORT_LINUX ?= /dev/ttyUSB0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbserial*)))
+
+# setup serial terminal
+#include $(RIOTMAKE)/tools/serial.inc.mk
+
+# Set default flash tool
+#PROGRAMMER ?= cc2538-bsl
+
+# For backward compatibility
+#ifneq (,$(PORT_BSL))
+ # $(warning Warning! PORT_BSL is deprecated use PROG_DEV)
+ # PROG_DEV ?= $(PORT_BSL)
+#endif
+
+#ifeq ($(PROGRAMMER),cc2538-bsl)
+ # FLASHFILE ?= $(HEXFILE)
+ # FFLAGS_OPTS ?= --bootloader-invert-lines
+ # PROG_BAUD ?= 460800
+ # include $(RIOTMAKE)/tools/cc2538-bsl.inc.mk
+#else ifeq ($(PROGRAMMER),jlink)
+  # Special flashing and reset scripts are required due to board hardware
+ # FLASH_ADDR = 0x200000
+ # JLINK_DEVICE = CC2538SF53
+ # JLINK_IF = JTAG
+ # JLINK_RESET_FILE = $(RIOTBOARD)/OMote/dist/hw_reset.seg
+ # include $(RIOTMAKE)/tools/jlink.inc.mk
+#endif
```
those are commented out - just drop that if it's not used.

benpicco commented on boards/OMote/Makefile.dep:
```diff
@@ -0,0 +1,8 @@
+
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+  USEMODULE += si7006
```
You didn't list this in your board feature list - is that sensor really on the board or is it a copy & paste artifact?

benpicco commented on boards/OMote/dist/reset.sh:
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# This script resets a CC2538SF53 target using JLink called
+# with a pre-defined reset sequence.
+
+# @author Hauke Petersen <hauke.petersen@fu-berlin.de>
+
+BINDIR=$1
+
+# create JLink command file for resetting the board
+echo "r" >> $BINDIR/reset.seg
+echo "g" >> $BINDIR/reset.seg
+echo "exit" >> $BINDIR/reset.seg
+
+# reset the board
+JLinkExe -device CC2538SF53 < $BINDIR/reset.seg
+echo ""
```
Instead of copying this, you can just use `boards/common/remote/dist/reset.sh`

benpicco commented on boards/OMote/board.c:
```diff
@@ -0,0 +1,65 @@
+{
+    gpio_init(LED0_PIN, GPIO_OUT);
+    gpio_init(LED1_PIN, GPIO_OUT);
+
```
```suggestion
```

benpicco commented on boards/OMote/Makefile.include:
```diff
@@ -0,0 +1,92 @@
+  FFLAGS  = $(BINDIR) $(FLASHFILE)
+  DEBUGGER = $(RIOTBOARD)/OMote/dist/debug.sh
+  DEBUGSERVER = JLinkGDBServer -device CC2538SF53
+  RESET ?= $(RIOTBOARD)/OMote/dist/reset.sh
```
```suggestion
  RESET ?= $(RIOTBOARD)/common/remote/dist/reset.sh
```

should work too.

benpicco commented on boards/OMote/Makefile.include:
```diff
@@ -0,0 +1,92 @@
+else ifeq ($(PROGRAMMER),jlink)
+  FLASHER = $(RIOTBOARD)/OMote/dist/flash.sh
+  FFLAGS  = $(BINDIR) $(FLASHFILE)
+  DEBUGGER = $(RIOTBOARD)/OMote/dist/debug.sh
```
```suggestion
  DEBUGGER = $(RIOTBOARD)/common/remote/dist/debug.sh
```

benpicco commented on boards/OMote/Makefile.include:
```diff
@@ -0,0 +1,92 @@
+  PROG_BAUD ?= 115200
+  include $(RIOTMAKE)/tools/cc2538-bsl.inc.mk
+else ifeq ($(PROGRAMMER),jlink)
+  FLASHER = $(RIOTBOARD)/OMote/dist/flash.sh
```
```suggestion
  FLASHER = $(RIOTBOARD)/common/remote/dist/flash.sh
```

benpicco commented on boards/omote/Makefile.include:
```diff
@@ -0,0 +1,35 @@
+PORT_LINUX  ?= /dev/ttyUSB0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
+
+# define the default flash-tool
+PROGRAMMER ?= cc2538-bsl
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+# For backward compatibility
+ifneq (,$(PORT_BSL))
+  $(warning Warning! PORT_BSL is deprecated use PROG_DEV)
+  PROG_DEV ?= $(PORT_BSL)
+endif
+
+ifeq ($(PROGRAMMER),cc2538-bsl)
+  PROG_BAUD ?= 115200
+  include $(RIOTMAKE)/tools/cc2538-bsl.inc.mk
+else ifeq ($(PROGRAMMER),jlink)
+  FLASHER = $(RIOTBOARD)/cc2538dk/dist/flash.sh
+  FFLAGS  = $(BINDIR) $(FLASHFILE)
+  DEBUGGER = $(RIOTBOARD)/cc2538dk/dist/debug.sh
+  DEBUGSERVER = JLinkGDBServer -device CC2538SF53
+  RESET ?= $(RIOTBOARD)/cc2538dk/dist/reset.sh
+  RESET_FLAGS ?= $(BINDIR)
+endif
+
+OFLAGS = --gap-fill 0xff
+FLASHFILE ?= $(BINFILE)
+DEBUGGER_FLAGS = $(BINDIR) $(ELFFILE)
+OBJDUMPFLAGS += --disassemble --source --disassembler-options=force-thumb
+
+# include common cc2538 includes
+INCLUDES += -I$(RIOTBOARD)/common/cc2538/include
```
Actually you should be able to drop all that and replace it with

```Makefile
include $(RIOTBOARD)/common/remote/Makefile.include
```

see #14554

benpicco replied:
Can you try this?

```suggestion
include $(RIOTBOARD)/common/remote/Makefile.include
```

Oppila-ms commented:
adding omote platform

benpicco commented on boards/omote/README.md:
```diff
@@ -0,0 +1,94 @@
+
+More Reading
+============
+1. [Oppila Microsystems omote website - http://www.oppila.in]
```
```suggestion
1. [Oppila Microsystems omote website](http://www.oppila.in)
```

jia200x commented:
Hi,

> Hello
>
> We have updated with contribution and testing procedures. I would like to know next procedures.>
>
> Regards
> Manikandan M

Great.
So, we keep this PR for reviews and new patches.

Next steps is to address comments by the community (@benpicco already [made some](https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r456887872) and they are still not addressed).
All changes should be committed and pushed to this branch.

Oppila-ms commented:
Hi

I would like to know the status of omote porting. 

Regards
Manikandan M

⁣Get BlueMail for Android ​

On 30 Jul 2020, 2:28 pm, at 2:28 pm, "José Alamos" <notifications@github.com> wrote:
>Hi,
>
>> Hello
>>
>> We have updated with contribution and testing procedures. I would
>like to know next procedures.>
>>
>> Regards
>> Manikandan M
>
>Great.
>So, we keep this PR for reviews and new patches.
>
>Next steps is to address comments by the community (@benpicco already
>[made
>some](https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r456887872)
>and they are still not addressed).
>All changes should be committed and pushed to this branch.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-666241304

benpicco commented:
It seems like you are only using the eMail interface.
Please open the Pull Request in a browser: https://github.com/RIOT-OS/RIOT/pull/14552

You should then see the open points. 

E.g. I think most of `Makefile.include` can be replaced by an already existing common Makefile already used by other CC2538 boards. Please check if this works for you.

benpicco commented on boards/omote/README.md:
```diff
@@ -0,0 +1,94 @@
+
+
+### For the CC2538EM (USB CDC-ACM)
+The omote has built-in support for USB 2.0 USB, Vendor and Product IDs are the following:
+
+  * VID 0x0451
+  * PID 0x16C8
```
You should add

```C
#define INTERNAL_PERIPHERAL_VID         (0x0451)
#define INTERNAL_PERIPHERAL_PID         (0x16C8)
```

to `board.h` if you want to automatically use them.

jia200x replied:
@Oppila-ms this was not addressed. Could you put this lines in `board.h`?

benpicco commented on boards/omote/board.c:
```diff
@@ -0,0 +1,64 @@
+ */
+static void rf_switch_init(void)
+{
+    /* Set RF 2.4GHz as default */
+    gpio_init(RF_SWITCH_GPIO, GPIO_OUT);
+    RF_SWITCH_2_4_GHZ;
```
There is only one radio on the board, so what does this switch do?

benpicco commented on boards/omote/Makefile.dep:
```diff
@@ -0,0 +1,3 @@
+ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
+  USEMODULE += cc2538_rf
+endif
```
You can ad

```
ifneq (,$(filter saul_default,$(USEMODULE)))
  USEMODULE += adxl345
  USEMODULE += bmp180
  USEMODULE += saul_gpio
endif
```

benpicco commented on boards/omote/board.c:
```diff
@@ -0,0 +1,63 @@
+
+/**
+ * @brief Initialize the Radio interface
+ * The omote features an on-board RF switch.
```
What does the switch switch?
Is there an on-board and an external antenna?

Oppila-ms commented:
Hi

I have removed a word switch and commited changes 

Regards
Ranjitha G Hiremath

On 30 Jul 2020, 11:42 pm, at 11:42 pm, benpicco <notifications@github.com> wrote:
>@benpicco commented on this pull request.
>
>
>
>> +
>+/**
>+ * @brief Initialize the boards on-board LEDs
>+ *
>+ * The LED initialization is hard-coded in this function. As the LEDs
>are
>+ * soldered onto the board they are fixed to their CPU pins.
>+ */
>+static inline void leds_init(void)
>+{
>+    gpio_init(LED0_PIN, GPIO_OUT);
>+    gpio_init(LED1_PIN, GPIO_OUT);
>+}
>+
>+/**
>+ * @brief Initialize the Radio interface
>+ * The omote features an on-board RF switch.
>
>What does the switch switch?
>Is there an on-board and an external antenna? 
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#pullrequestreview-458641380

Oppila-ms commented:
Hi

Could you please help me know the status of omote porting.

Regards
Manikandan M

On 30 Jul 2020, 11:42 pm, at 11:42 pm, benpicco <notifications@github.com> wrote:
>@benpicco commented on this pull request.
>
>
>
>> +
>+/**
>+ * @brief Initialize the boards on-board LEDs
>+ *
>+ * The LED initialization is hard-coded in this function. As the LEDs
>are
>+ * soldered onto the board they are fixed to their CPU pins.
>+ */
>+static inline void leds_init(void)
>+{
>+    gpio_init(LED0_PIN, GPIO_OUT);
>+    gpio_init(LED1_PIN, GPIO_OUT);
>+}
>+
>+/**
>+ * @brief Initialize the Radio interface
>+ * The omote features an on-board RF switch.
>
>What does the switch switch?
>Is there an on-board and an external antenna? 
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#pullrequestreview-458641380

jia200x commented:
@Oppila-ms AFAIS https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r461448092 was not addressed

jia200x commented:
@Oppila-ms otherwise, it seems most of the comments by @benpicco were addressed. 

After all changes are addressed, we can activate the CI and see if more changes are required. After that, adapt the commit messages to the RIOT format and squash.
With all these steps we should be able to merge it to the master branch

jia200x commented on boards/omote/README.md:
```diff
@@ -0,0 +1,73 @@
+Oppila Microsystems omote platform -  http://www.oppila.in
+==========================================================
+
+The omote platform  is a IoT Hardware development platform based
```
```suggestion
The omote platform is a IoT Hardware development platform based
```

jia200x commented on boards/omote/README.md:
```diff
@@ -0,0 +1,73 @@
+
+Once all drivers have been installed correctly:
+
+On windows, devices will appear as a virtual `COM` port.
```
```suggestion
On Windows, devices will appear as a virtual `COM` port.
```

jia200x commented on boards/omote/README.md:
```diff
@@ -0,0 +1,73 @@
+
+On Linux, devices will appear under `/dev/`.
+
+On OS X, `/dev/tty.SLAB_USBtoUARTx`.
```
```suggestion
On macOS, `/dev/tty.SLAB_USBtoUARTx`.
```

jia200x commented on boards/omote/board.c:
```diff
@@ -0,0 +1,63 @@
+/*
+ *Copyright (C) 2020 Oppila Microsystems - http://www.oppila.in
```
```suggestion
 * Copyright (C) 2020 Oppila Microsystems - http://www.oppila.in
```

jia200x commented on boards/omote/board.c:
```diff
@@ -0,0 +1,63 @@
+{
+    /* Set RF 2.4GHz */
+    gpio_init(RF_GPIO, GPIO_OUT);
+    RF_2_4_GHZ;
```
It's not clear what this macro does or why it is there. Could you add documentation regarding this? And maybe this macro is not needed if it's simply a `gpio_clear` call

jia200x commented on boards/omote/doc.txt:
```diff
@@ -0,0 +1,48 @@
+Bash
+$ make BOARD=omote flash
+
+The omote has 13 burgs pinouts to connect boolean,digital and analog sensors based on I2C,UART,and SPI Protocols as well as other sensors or actuators you may need to connect.
```
what does `burgs` mean here?

benpicco commented on boards/omote/Makefile.features:
```diff
@@ -4,7 +4,6 @@ CPU_MODEL = cc2538sf53
 # Put defined MCU peripherals here (in alphabetical order)
 FEATURES_PROVIDED += periph_adc
 FEATURES_PROVIDED += periph_i2c
-FEATURES_PROVIDED += periph_rtt
```
Why remove that?

benpicco commented on boards/omote/board.c:
```diff
@@ -12,26 +12,21 @@
     /* initialize the CPU */
     cpu_init();
-    /* initialize the 2.4GHz RF */
-    rf_init();
```
Does the radio still work with this removed?

aabadie commented on boards/omote/README.md:
```diff
@@ -0,0 +1,73 @@
+Oppila Microsystems omote platform -  http://www.oppila.in
```
Please move the content of the README.md file to doc.txt. It contains valuable information to have in the generated documentation.
Once its content is moved, simply remove the readme.

Oppila-ms commented:
Hi

Yes it works by removing that as we have only one rf i.e 2.4GHz and we removed rtt because we don't have external rtc on our board. So do we have to mention it if we don't have external rtc ..?

Regards
Manikandan Mi

On 3 Aug 2020, 6:11 pm, at 6:11 pm, benpicco <notifications@github.com> wrote:
>@benpicco commented on this pull request.
>
>
>
>> -    /* initialize the 2.4GHz RF */
>-    rf_init();
>
>Does the radio still work with this removed? 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#pullrequestreview-459992233

benpicco commented:
> we removed rtt because we don't have external rtc on our board. 

I guess you mean you don't have a 32kHz crystal for the RTT?

Oppila-ms commented:
Hi

We have 32KHz crystal oscillator which is connected to microcontroller and i have added rtt in makefile.features

Regards
Manikandan M

On 3 Aug 2020, 7:07 pm, at 7:07 pm, benpicco <notifications@github.com> wrote:
>> we removed rtt because we don't have external rtc on our board. 
>
>I guess you mean you don't have a 32kHz crystal for the RTT?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-668027075

Oppila-ms commented:
We have ported omote yesterday and waiting for the status. Could you please help me know the status.


On 3 Aug 2020, 7:07 pm, at 7:07 pm, benpicco <notifications@github.com> wrote:
>> we removed rtt because we don't have external rtc on our board. 
>
>I guess you mean you don't have a 32kHz crystal for the RTT?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-668027075

Oppila-ms commented:
Hi

Could you please help us know the status of omote porting. We are waiting for your reply to know the status.

Regards
Manikandan M

On 3 Aug 2020, 7:07 pm, at 7:07 pm, benpicco <notifications@github.com> wrote:
>> we removed rtt because we don't have external rtc on our board. 
>
>I guess you mean you don't have a 32kHz crystal for the RTT?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-668027075

benpicco commented on boards/omote/include/board.h:
```diff
@@ -0,0 +1,95 @@
+ * @name  RF Pin definition
+ * @{
+ */
+#define RF_GPIO      GPIO_PD2
```
What does this pin do?

benpicco commented:
Can you please try the suggestion from https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r461448092 ?
It should make the code much simpler.

benpicco commented:
Great!
Can you post the output of `tests/driver_adxl345` and `tests/driver_bmp180` to see if those are configured correctly?

benpicco commented on boards/omote/board.c:
```diff
@@ -0,0 +1,45 @@
+    /* The boot pin must be set to input otherwise it may lock the bootloader */
+    gpio_init(BOOT_PIN, GPIO_IN);
+
+    gpio_init(BTN0_PIN, BTN0_MODE);
```
An application using the button will take care of that (and might set up a callback) - you can drop this.

aabadie replied:
This is still unaddressed.

jia200x replied:
@Oppila-ms please address this one too.

Oppila-ms commented:
Actually that is a user button. We are setting  pin mode as input mode and initialising it.


On 5 Aug 2020, 4:11 pm, at 4:11 pm, benpicco <notifications@github.com> wrote:
>@benpicco commented on this pull request.
>
>
>
>> +
>+#include "board.h"
>+#include "cpu.h"
>+
>+static inline void leds_init(void);
>+
>+void board_init(void)
>+{
>+    /* initialize the boards LEDs */
>+    leds_init();
>+    /* initialize the CPU */
>+    cpu_init();
>+    /* The boot pin must be set to input otherwise it may lock the
>bootloader */
>+    gpio_init(BOOT_PIN, GPIO_IN);
>+
>+    gpio_init(BTN0_PIN, BTN0_MODE);
>
>An application using the button will take care of that (and might set
>up a callback) - you can drop this. 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#pullrequestreview-461543045

benpicco commented:
Yes but usually applications will take care of initializing it.
e.g. SAUL will initialize the button GPIO or when you run `tests/buttons` it should also still work without init in`board.c`.
But if you want you can keep it.

Oppila-ms commented:
Hi

Please find the below attachments of adxl345 & bmp180 sensors output

On 5 Aug 2020, 4:10 pm, at 4:10 pm, benpicco <notifications@github.com> wrote:
>Great!
>Can you post the output of `tests/driver_adxl345` and
>`tests/driver_bmp180` to see if those are configured correctly? 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669119822

benpicco commented:
The email -> GitHub gateway did strip the attachments it seems.

Oppila-ms commented:
> Great!
> Can you post the output of `tests/driver_adxl345` and `tests/driver_bmp180` to see if those are configured correctly?

[output.zip](https://github.com/RIOT-OS/RIOT/files/5029556/output.zip)
output of adxl345 and bmp180

benpicco commented:
Great!
Please squash. (You can do `git rebase -i master` and squash all commits into one.)
Please use a commit message that matches the PR description. (`boards/omote: add support for OMote board` )

Oppila-ms commented:
I have squashed one commit. 


On 5 Aug 2020, 9:31 pm, at 9:31 pm, benpicco <notifications@github.com> wrote:
>Great!
>Please squash. (You can do `git rebase -i master` and squash all
>commits into one.)
>Please use a commit message that matches the PR description.
>(`boards/omote: add support for OMote board` )
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669277730

benpicco commented:
You will also have to push :wink:

Oppila-ms commented:
> You will also have to push wink

should i checkout to our branch

benpicco commented:
Huh?
Which branch did you rebase?

If you are on the `oppila` branch, a `git push -f` should be sufficient.

Oppila-ms commented:
> Huh?
> Which branch did you rebase?
> 
> If you are on the `oppila` branch, a `git push -f` should be sufficient.

this is the output i am getting 
boards/omote/README.md: needs merge
error: you need to resolve your current index first

jia200x commented:
@Oppila-ms could you paste the output of:
```
git status
```
?

Oppila-ms commented:
> @Oppila-ms could you paste the output of:
> 
> ```
> git status
> ```
> 
> ?

interactive rebase in progress; onto 73351c507
Last command done (1 command done):
   pick 600d21478 updating omote
Next commands to do (9 remaining commands):
   pick 4e2c372a8 Readme file changed
   pick f60f4e47b Readme.md and doc file modified
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'oppila' on '73351c507'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

	both modified:   boards/omote/README.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	tests/driver_adxl345/output

no changes added to commit (use "git add" and/or "git commit -a")

Oppila-ms commented:
> > @Oppila-ms could you paste the output of:
> > ```
> > git status
> > ```
> > 
> > 
> > ?
> 
> interactive rebase in progress; onto [73351c5](https://github.com/RIOT-OS/RIOT/commit/73351c507a8e9ed188349aa8cac4ebe6dc3a687c)
> Last command done (1 command done):
> pick [600d214](https://github.com/RIOT-OS/RIOT/commit/600d21478989b771a1147df33db82755320e94cd) updating omote
> Next commands to do (9 remaining commands):
> pick [4e2c372](https://github.com/RIOT-OS/RIOT/commit/4e2c372a8100efe19ce0fd9eaa68028fe962921f) Readme file changed
> pick [f60f4e4](https://github.com/RIOT-OS/RIOT/commit/f60f4e47bebd8b4b044071714ff6a9556ae3556f) Readme.md and doc file modified
> (use "git rebase --edit-todo" to view and edit)
> You are currently rebasing branch 'oppila' on '73351c507'.
> (fix conflicts and then run "git rebase --continue")
> (use "git rebase --skip" to skip this patch)
> (use "git rebase --abort" to check out the original branch)
> 
> Unmerged paths:
> (use "git reset HEAD ..." to unstage)
> (use "git add ..." to mark resolution)
> 
> ```
> both modified:   boards/omote/README.md
> ```
> 
> Untracked files:
> (use "git add ..." to include in what will be committed)
> 
> ```
> tests/driver_adxl345/output
> ```
> 
> no changes added to commit (use "git add" and/or "git commit -a")

could you help me to proceed further

jia200x commented:
@Oppila-ms you are in the middle of a rebase. Please follow these steps:

1) `git rebase --abort`
2) Make sure you are in the right branch (if not, switch to the `oppila` branch)
3) `git rebase -i master`
4) Use the `reword` command for the first commit. For all the others, use the `fixup` command. It should look like this:

```
r 1a56229a00 boards/OMote
f 223345d0bc boards/omote
f 08b217cff6 removed dist file
f e4cd92073f modified
f 4996ab7b08 blank lines are removed
f 4bdd89494c modified doc file
f 5fff8eb8da doc file corrected
f 4713394792 board.h file corrected
f 600d214789 updating omote
f 4e2c372a81 Readme file changed
f f60f4e47be Readme.md and doc file modified
f cf7b975e63 omote
f 857b621f59 board.c file
f 47020d7a36 board.c switch removed
f 3505a59c52 board.h whitespace
f 22bb9c3536 omote
f c1f0a12d30 rtt added
f 6502c57a5e Makefile.include modified

# Rebase ecab75b529..6502c57a5e onto 22bb9c3536 (18 commands)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup <commit> = like "squash", but discard this commit's log message
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified). Use -c <commit> to reword the commit message.
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
```

5) Save the file and exit
6) Choose the name for the single commit (e.g `boards/omote: add support for OMote board)
7) Save file and exit
8) `git push -f`

aabadie commented on boards/omote/Makefile.dep:
```diff
@@ -0,0 +1,9 @@
+ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
```
IIUC, `netdev_default` should be enough. Do you confirm @benpicco ?

```suggestion
ifneq (,$(filter netdev_default,$(USEMODULE)))
```

aabadie commented on boards/omote/doc.txt:
```diff
@@ -0,0 +1,99 @@
+The omote has built-in support to flash over USB using the BSL.It requires to unlock the bootloader by manually pressing the `BSL
+button` and `reset button`
+
+Bash
```
This won't render to a code block. You can rebase your branch on top of a more recent master version and CircleCI will provide a preview of the generated documentation.

Oppila-ms replied:
> This won't render to a code block. You can rebase your branch on top of a more recent master version and CircleCI will provide a preview of the generated documentation.

could not understand what i have to do sir

jia200x replied:
It's because of Markdown syntax. In order to render, please use the following syntax with triple back-quotes:

\`\`\`bash
your bash commands
\`\`\`

E.g:

```bash
make BOARD=omote flash
```

jia200x replied:
@Oppila-ms the syntax is still wrong.
Please copy __exactly__ this lines:

\`\`\`bash
make BOARD=omote flash
\`\`\`

Oppila-ms replied:
> @Oppila-ms the syntax is still wrong.
> Please copy **exactly** this lines:
> 
> ```bash
> make BOARD=omote flash
> ```

sir i copied the same lines and pushed.

jia200x replied:
great. I will check if there are pending reviews

Oppila-ms commented:
> > @Oppila-ms could you paste the output of:
> > ```
> > git status
> > ```
> > 
> > 
> > ?
> 
> interactive rebase in progress; onto [73351c5](https://github.com/RIOT-OS/RIOT/commit/73351c507a8e9ed188349aa8cac4ebe6dc3a687c)
> Last command done (1 command done):
> pick [600d214](https://github.com/RIOT-OS/RIOT/commit/600d21478989b771a1147df33db82755320e94cd) updating omote
> Next commands to do (9 remaining commands):
> pick [4e2c372](https://github.com/RIOT-OS/RIOT/commit/4e2c372a8100efe19ce0fd9eaa68028fe962921f) Readme file changed
> pick [f60f4e4](https://github.com/RIOT-OS/RIOT/commit/f60f4e47bebd8b4b044071714ff6a9556ae3556f) Readme.md and doc file modified
> (use "git rebase --edit-todo" to view and edit)
> You are currently rebasing branch 'oppila' on '73351c507'.
> (fix conflicts and then run "git rebase --continue")
> (use "git rebase --skip" to skip this patch)
> (use "git rebase --abort" to check out the original branch)
> 
> Unmerged paths:
> (use "git reset HEAD ..." to unstage)
> (use "git add ..." to mark resolution)
> 
> ```
> both modified:   boards/omote/README.md
> ```
> 
> Untracked files:
> (use "git add ..." to include in what will be committed)
> 
> ```
> tests/driver_adxl345/output
> ```
> 
> no changes added to commit (use "git add" and/or "git commit -a")

could you help me to proceed further

> @Oppila-ms you are in the middle of a rebase. Please follow these steps:
> 
>     1. `git rebase --abort`
> 
>     2. Make sure you are in the right branch (if not, switch to the `oppila` branch)
> 
>     3. `git rebase -i master`
> 
>     4. Use the `reword` command for the first commit. For all the others, use the `fixup` command. It should look like this:
> 
> 
> ```
> r 1a56229a00 boards/OMote
> f 223345d0bc boards/omote
> f 08b217cff6 removed dist file
> f e4cd92073f modified
> f 4996ab7b08 blank lines are removed
> f 4bdd89494c modified doc file
> f 5fff8eb8da doc file corrected
> f 4713394792 board.h file corrected
> f 600d214789 updating omote
> f 4e2c372a81 Readme file changed
> f f60f4e47be Readme.md and doc file modified
> f cf7b975e63 omote
> f 857b621f59 board.c file
> f 47020d7a36 board.c switch removed
> f 3505a59c52 board.h whitespace
> f 22bb9c3536 omote
> f c1f0a12d30 rtt added
> f 6502c57a5e Makefile.include modified
> 
> # Rebase ecab75b529..6502c57a5e onto 22bb9c3536 (18 commands)
> #
> # Commands:
> # p, pick <commit> = use commit
> # r, reword <commit> = use commit, but edit the commit message
> # e, edit <commit> = use commit, but stop for amending
> # s, squash <commit> = use commit, but meld into previous commit
> # f, fixup <commit> = like "squash", but discard this commit's log message
> # x, exec <command> = run command (the rest of the line) using shell
> # b, break = stop here (continue rebase later with 'git rebase --continue')
> # d, drop <commit> = remove commit
> # l, label <label> = label current HEAD with a name
> # t, reset <label> = reset HEAD to a label
> # m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
> # .       create a merge commit using the original merge commit's
> # .       message (or the oneline, if no original merge commit was
> # .       specified). Use -c <commit> to reword the commit message.
> #
> # These lines can be re-ordered; they are executed from top to bottom.
> #
> # If you remove a line here THAT COMMIT WILL BE LOST.
> #
> # However, if you remove everything, the rebase will be aborted.
> #
> ```
> 
>     1. Save the file and exit
> 
>     2. Choose the name for the single commit (e.g `boards/omote: add support for OMote board`)
> 
>     3. Save file and exit
> 
>     4. `git push -f`

sir thank you so much for your help and i have done according your good suggestions. could you please check out sir.

jia200x commented:
@Oppila-ms there are some comments that were still not addressed. Could you check them first?

- https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466302119
- https://github.com/RIOT-OS/RIOT/pull/14552/files/6502c57a5ebbeb0b2953147b7153bab7c79509b7#diff-8a5cc0c3c897f89e41a45f3a7c2bd0c2

Also, it seems you forgot to push

Oppila-ms commented:
This we would like to keep it so we dud not address that

On 6 Aug 2020, 5:31 pm, at 5:31 pm, "José Alamos" <notifications@github.com> wrote:
>@jia200x commented on this pull request.
>
>
>
>> +
>+#include "board.h"
>+#include "cpu.h"
>+
>+static inline void leds_init(void);
>+
>+void board_init(void)
>+{
>+    /* initialize the boards LEDs */
>+    leds_init();
>+    /* initialize the CPU */
>+    cpu_init();
>+    /* The boot pin must be set to input otherwise it may lock the
>bootloader */
>+    gpio_init(BOOT_PIN, GPIO_IN);
>+
>+    gpio_init(BTN0_PIN, BTN0_MODE);
>
>@Oppila-ms please address this one too.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466361733

jia200x commented:
> This we would like to keep it so we dud not address that

@benpicco @aabadie I don't have an opinion on that topic. How is it usually handled in other boards? What do you think?

benpicco commented:
There are indeed some boards that init the button in `board_init()`.
Thinking about it that might actually make sense from an energy perspective, depending on how the button / hardware is set up.

Oppila-ms commented:
> There are indeed some boards that init the button in `board_init()`.
> Thinking about it that might actually make sense from an energy perspective, depending on how the button / hardware is set up.

Oppila-ms commented:
> > There are indeed some boards that init the button in `board_init()`.
> > Thinking about it that might actually make sense from an energy perspective, depending on how the button / hardware is set up.

jia200x commented:
> There are indeed some boards that init the button in board_init().
Thinking about it that might actually make sense from an energy perspective, depending on how the button / hardware is set up.

So, would it be OK as it is now?

benpicco commented:
I'd say so.

Oppila-ms commented:
sir i gave git push -f also

jia200x commented:
2 more things and it seems we should be ready to continue with the CI.

1. Please address https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466428770
2. It seems something went wrong with the rebase. Did you follow all the steps in https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669806309? The result should be just one single commit with message `boards/omote: add support OMote board`, not 18 commits as Github shows. Did you forget to change those `pick` command to one `reword` and all the others to `fixup`?

Oppila-ms commented:
> 2 more things and it seems we should be ready to continue with the CI.
> 
>     1. Please address [#14552 (comment)](https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466428770)
> 
>     2. It seems something went wrong with the rebase. Did you follow all the steps in [#14552 (comment)](https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669806309)? The result should be just one single commit with message `boards/omote: add support OMote board`, not 18 commits as Github shows. Did you forget to change those `pick` command to one `reword` and all the others to `fixup`?

yes i did as you directed me but those commits are not changing up and when i give push it is giving error. before this once back i merged so i think because of that it is not updating.

Oppila-ms commented:
last two commits are not updating even. i am struck to proceed further sir could you please direct me

jia200x commented:
it seems you literally copied the step 4). The idea was to have the first commit with "reword" and the other with "fixup". Since that is an outdated version (you added some changes in the middle), those commits are "lost" now.

In order to recover them, please use the `git reflog` command and find where was the point where you did the last rebase (you can directly point to the hash of the last commit).

when you know the hash, make the branch point to that commit (`git reset --hard <commit_hash_of_last_missed_commit>`) and `git push -f` again.

after you do that, please address again https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466428770 (probably that commit will be lost) and report back.

If you want, I can squash it and leave your authorship in the squashed commits. But in order to do that, you would need to follow the procedure described in this comment first

jia200x commented:
(if you get lost during https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669957422, please send the status of `git reflog` and we can write down the required commands)

Oppila-ms commented:
I have removed readme file so i did not add those lines in board.h file sir

⁣Get BlueMail for Android ​

On 6 Aug 2020, 7:24 pm, at 7:24 pm, "José Alamos" <notifications@github.com> wrote:
>@jia200x commented on this pull request.
>
>
>
>> +The omote has built-in support for USB 2.0 USB, Vendor and Product
>IDs are the following:
>+
>+  * VID 0x0451
>+  * PID 0x16C8
>
>@Oppila-ms this was not addressed. Could you put this lines in
>`board.h`?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466428770

jia200x commented:
> I have removed readme file so i did not add those lines in board.h file sir

But those lines would enable USB automatically. If they are not there, the user would need to manually specify them

Oppila-ms commented:
> (if you get lost during [#14552 (comment)](https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669957422), please send the status of `git reflog` and we can write down the required commands)

output of git reflog
(HEAD, origin/master, origin/HEAD, master) HEAD@{0}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{1}: rebase -i (abort): updating HEAD
6502c57a5 (origin/oppila, oppila) HEAD@{2}: rebase: aborting
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{3}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{4}: rebase -i (abort): updating HEAD
6502c57a5 (origin/oppila, oppila) HEAD@{5}: rebase: aborting
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{6}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{7}: rebase -i (abort): updating HEAD
57b385112 HEAD@{8}: rebase: aborting
57b385112 HEAD@{9}: commit: doc file changed
55eef41d2 HEAD@{10}: commit: updated omote
6502c57a5 (origin/oppila, oppila) HEAD@{11}: checkout: moving from 73351c507a8e9ed188349aa8cac4ebe6dc3a687c to oppila
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{12}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{13}: rebase -i (abort): updating HEAD
6502c57a5 (origin/oppila, oppila) HEAD@{14}: rebase: aborting
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{15}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{16}: checkout: moving from oppila to oppila
6502c57a5 (origin/oppila, oppila) HEAD@{17}: rebase -i (abort): updating HEAD
6502c57a5 (origin/oppila, oppila) HEAD@{18}: rebase: aborting
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{19}: rebase -i (start): checkout master
6502c57a5 (origin/oppila, oppila) HEAD@{20}: commit: Makefile.include modified
c1f0a12d3 HEAD@{21}: commit: rtt added
22bb9c353 HEAD@{22}: commit: omote
3505a59c5 HEAD@{23}: commit: board.h whitespace
47020d7a3 HEAD@{24}: commit: board.c switch removed
857b621f5 HEAD@{25}: commit: board.c file
cf7b975e6 HEAD@{26}: commit: omote
f60f4e47b HEAD@{27}: commit: Readme.md and doc file modified
4e2c372a8 HEAD@{28}: commit: Readme file changed
600d21478 HEAD@{29}: checkout: moving from master to oppila
73351c507 (HEAD, origin/master, origin/HEAD, master) HEAD@{30}: commit: update omote
54a0814b1 (origin/Oppilams) HEAD@{31}: checkout: moving from oppila to master
600d21478 HEAD@{32}: commit: updating omote
471339479 HEAD@{33}: checkout: moving from master to oppila
54a0814b1 (origin/Oppilams) HEAD@{34}: merge Oppilams: Fast-forward
471339479 HEAD@{35}: checkout: moving from Oppilams to master
54a0814b1 (origin/Oppilams) HEAD@{36}: commit: borads/omote file
:

jia200x commented:
@Oppila-ms 

Please run the following commands in your branch (`oppila`):
1. `git rebase --abort`
2. `git reset --hard 57b385112`
3. `git push -f`

Then, please address https://github.com/RIOT-OS/RIOT/pull/14552#discussion_r466428770 and push it again (without `-f`).
When that's done, I can take over the rebase process

jia200x commented:
ok, I will take over

Oppila-ms commented:
> ok, I will take over

thank you so much

jia200x commented:
see https://github.com/RIOT-OS/RIOT/pull/14720. Let's keep this one open for now

benpicco commented:
We could have also done a "Squash and Merge" in the GitHub GUI.
Anyway, let's go on with the new PR then.

Oppila-ms commented:
Okay 

On 6 Aug 2020, 8:55 pm, at 8:55 pm, benpicco <notifications@github.com> wrote:
>We could have also done a "Squash and Merge" in the GitHub GUI.
>Anyway, let's go on with the new PR then.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669997189

Oppila-ms commented:
Can i know when this will be updated


On 6 Aug 2020, 8:55 pm, at 8:55 pm, benpicco <notifications@github.com> wrote:
>We could have also done a "Squash and Merge" in the GitHub GUI.
>Anyway, let's go on with the new PR then.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-669997189

benpicco commented:
superseded by #14720

Oppila-ms commented:
Thank you 

⁣But i have some questions to ask if you don't mind can you answer me

On 7 Aug 2020, 6:40 pm, at 6:40 pm, benpicco <notifications@github.com> wrote:
>superseded by #14720 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-670507929

Oppila-ms commented:
If we want to update anything in our omote means do we have to fork again and do a pull request . Or how can we  update any changes with our board

Regards
Manikandan M

On 7 Aug 2020, 6:40 pm, at 6:40 pm, benpicco <notifications@github.com> wrote:
>superseded by #14720 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-670507929

benpicco commented:
Sure, you can just create a new branch off the master branch now, it will have your board in it.

Oppila-ms commented:
Okay, thanks to all your team



On 10 Aug 2020, 5:26 pm, at 5:26 pm, benpicco <notifications@github.com> wrote:
>Sure, you can just create a new branch off the master branch now, it
>will have your board in it.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/RIOT-OS/RIOT/pull/14552#issuecomment-671311640
